wallet/src/main/scala/org/bitcoins/wallet/LockedWallet.scala

```diff
@@ -161,11 +182,160 @@ abstract class LockedWallet extends LockedWalletApi with BitcoinSLogger {
     }
   }
 
+  private case class OutputWithIndex(output: TransactionOutput, index: Int)
+
   /**
-    * @inheritdoc
+    * Processes an incoming transaction that's new to us
+    * @return `Some(tx)` if we found anything relevant to us
+    *         in the given transaction, `None` if nothing was
+    *         of relevance
+    */
+  private def processNewIncomingTx(
+      transaction: Transaction,
+      confirmations: Int): Future[Option[IncomingTransaction]] = {
+    addressDAO.findAll().flatMap { addrs =>
+      val relevantOutsWithIdx: Seq[OutputWithIndex] = {
+        val withIndex =
+          transaction.outputs.zipWithIndex
+        withIndex.collect {
+          case (out, idx)
+              if addrs.map(_.scriptPubKey).contains(out.scriptPubKey) =>
+            OutputWithIndex(out, idx)
+        }
+      }
+
+      relevantOutsWithIdx match {
```

Comment:
Overally I think this should be moved into a function that processes each relevantOuts and returns a `Vector[Future[IncomingUtxo]]` and then sequences those Futures to return a single `Future[Vector[IncomingUtxo]]`